components/ImageUpload.vue

```diff
@@ -1,7 +1,76 @@
-<script lang="ts" setup></script>
+<script lang="ts" setup>
 
-<template lang="pug">
+const emit = defineEmits([])
+
+const getSignedUrl = async (file: File) => {
+  // perform the request to get the presigned url from server (image_upload) and return it
+  // the api returns {uploadUrl, contentURl} from image_upload.post.ts
+  // upload url gets used to upload the file, the content URL is going to be used to 
+  // create an image entry in the database (This is a third api call to image.post.ts)
+  try {
+    const { data } = await useFetch('/api/image_upload', {
+        method: 'POST',
+        body: { size:file.size, contentType: file.type }
+      }).then((res) => res.json())
+
+      const { uploadUrl, contentUrl } = data
+
+      return { uploadUrl, contentUrl }
+    } catch (error) {
+      console.error(error)
+      throw new Error('Failed to get signed URL')
+  }
+}
+
+// actually uploads images using presigned url to s3 bucket
+const uploadFile = async (data, ContentLength, ContentType, presignedUrl, imageLink: string)  => {
+  const response = await fetch(presignedUrl, {
```

Comment:
You can actually make this the file upload handler, so that it takes the even and extracts the file information, then makes the fetch call - the arguments will be the incoming event instead of data etc. This function should make the call to getPresignedUrl, and then immediately use that information to do the upload